src/lightning/data/streaming/dataset.py

```diff
@@ -203,6 +192,50 @@ def __iter__(self) -> "StreamingDataset":
 
         return self
 
+    def _resume(self, chunks_replica: List[int], intervals_replica: List[Any]) -> None:
+        assert self._state_dict
+        assert self.worker_env
+        assert self.shuffler
+
+        restart_keys = sorted(self._state_dict)
+
+        # Get the state from the previous run
+        state: Dict[str, Any] = self._state_dict[restart_keys[-1]]
+
+        num_workers = state["num_workers"]
+        batch_size = state["batch_size"]
+
+        # TODO: Implement elastic sampling where the number of workers, ranks can change.
+        num_samples_yielded = sum([state["num_samples_yielded"] for state in self._state_dict.values()])
+
+        # replay sampling from each worker / chunks using the batch size
+        workers_chunks, workers_intervals = _associate_chunks_to_workers(
+            num_workers, self.worker_env, chunks_replica, intervals_replica
+        )
+        indexes = _replay_sampling(num_samples_yielded, batch_size, num_workers)
+        chunks_index, indexes = _replay_chunks_sampling(workers_intervals, indexes)
+
+        # select the chunks and intervals associated to this worker
+        worker_rank = self.worker_env.rank
+        self.num_chunks = len(workers_intervals[worker_rank])
+        self.chunk_index = chunks_index[worker_rank]
+        self.worker_chunks = workers_chunks[worker_rank]
+        self.worker_intervals = workers_intervals[worker_rank]
+
+        # replay the indexes for the current chunks
+        interval = workers_intervals[worker_rank][self.chunk_index]
+        current_indexes = np.arange(interval[0], interval[1])
+
+        # re-shuffle the indexes
+        current_indexes = self.shuffler(current_indexes, self.num_chunks, self.current_epoch, self.chunk_index)
+
+        # skip any indexes already consumed
+        current_indexes = current_indexes[indexes[worker_rank] :]
+        self.current_indexes = current_indexes
+
+        # bump the chunk_index
+        self.chunk_index += 1
```

Comment:
Just a clarification: here the chunk index is bumped because it's pointing to the next one? We could still be at some point of the previous chunk no?

Reply:
Yes, that's correct, I will probably change this in another PR to make it more clear. But right now, we are bumping when generating the indexes from a new chunk: https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/data/streaming/dataset.py#L240